runtime/compiler/CMakeLists.txt

```diff
@@ -334,7 +334,7 @@ omr_stringify(J9_CXXFLAGS_STR ${J9_SHAREDFLAGS} ${J9_CXXFLAGS})
 set(CMAKE_CXX_FLAGS "${J9_CXXFLAGS_STR}")
 set(CMAKE_C_FLAGS "${J9_CFLAGS_STR}")
 
-if(OMR_ARCH_X86 OR OMR_ARCH_S390 OR OMR_ARCH_AARCH64)
+if(OMR_ARCH_AARCH64 OR (OMR_ARCH_POWER AND NOT OMR_TOOLCONFIG STREQUAL "xlc") OR OMR_ARCH_S390 OR OMR_ARCH_X86)
```

Comment:
instead of identifying the compiler with **NOT** clause, i would suggest using predefined macro(s) offered by the compiler itself:  `__clang__`  (it is good for both xlclang and OpenXL)

Reply:
I did it this way because of a suggestion in an [earlier review](https://github.com/eclipse-openj9/openj9/pull/18614#discussion_r1601754894) from Keith, because I originally used `OMR_ARCH_POWER AND (__clang__ OR __GNUC__ OR __GNUG__)`.

Reply:
@zl-wang These are cmake variables, not compiler macros. Do you still have an objection, or do you approve?

Reply:
i can live with either. no strong opinion.

Reply:
Thank you; please formally approve this change.